app/services/db/monthly_sp_auth_count/new_unique_monthly_user_counts_by_partner.rb

```diff
@@ -43,31 +43,33 @@ def call(partner:, issuers:, start_date:, end_date:)
           ) do
             Reports::BaseReport.transaction_with_timeout do
               ActiveRecord::Base.connection.execute(query).each do |row|
-                user_id = row['user_id']
                 year_month = row['year_month']
                 profile_age = row['profile_age']
 
-                year_month_to_users_to_profile_age[year_month][user_id] = profile_age
+                user_unique_id = row['user_id'].to_s + '_' + row['profile_verified_at'].to_s +
+                                 '_' + row['profile_age'].to_s
```

Comment:
Ruby hashes can take anything as keys, so I would consider making this object of some sort. A few ideas:
1. A `Data` struct, these have equality built-in
    ```ruby
    UserVerifiedKey = Data.define(:user_id, :profile_verified_at, :profile_age)
    # ... 
    key = UserVerifiedKey.new(user_id:, profile_verified_at:, profile_at:)
    ```
2. A hash `{ user_id:, profile_verified_at:, profile_age: }`

Reply:
Implemented